python/tests/test_db.py

```diff
@@ -159,6 +159,43 @@ def test_delete_table(tmp_path):
     db.drop_table("does_not_exist", ignore_missing=True)
 
 
+def test_drop_database(tmp_path):
+    db = lancedb.connect(tmp_path)
+    data = pd.DataFrame(
+        {
+            "vector": [[3.1, 4.1], [5.9, 26.5]],
+            "item": ["foo", "bar"],
+            "price": [10.0, 20.0],
+        }
+    )
+    new_data = pd.DataFrame(
+        {
+            "vector": [[5.1, 4.1], [5.9, 10.5]],
+            "item": ["kiwi", "avocado"],
+            "price": [12.0, 17.0],
+        }
+    )
+    db.create_table("test", data=data)
+    with pytest.raises(Exception):
+        db.create_table("test", data=data)
+
+    assert db.table_names() == ["test"]
+
+    db.create_table("new_test", data=new_data)
+    assert db.table_names() == ["new_test", "test"]
+
+    db.drop_database()
+    assert db.table_names() == []
+
+    db.create_table("test", data=new_data)
+    db.drop_table("test")
+    assert db.table_names() == []
+
+    # dropping a empty database should pass
+    db.drop_database()
```

Comment:
Would you expect two consecutive calls to `drop_database` to pass?  I think that will fail because `delete_dir` expects the directory to exist.